scripts/bootstrap-aio.yml

```diff
@@ -0,0 +1,157 @@
+          result_code=2
+        fi
+        exit ${result_code}
+      register: user_variables_rename
+      changed_when: user_variables_rename.rc == 2
+      failed_when: user_variables_rename.rc not in [0, 2]
+
+    # TODO(odyssey4me): Switch this to use https://review.openstack.org/423368
+    # once it is merged upstream and available in RPC-O.
+    - name: Rename standard OSA user-space secret file
+      shell: |
+        result_code=0
+        if [ -e "/etc/openstack_deploy/user_secrets.yml" ]; then
+          mv /etc/openstack_deploy/user_secrets.yml /etc/openstack_deploy/user_osa_secrets.yml
+          result_code=2
+        fi
+        exit ${result_code}
+      register: user_secrets_rename
+      changed_when: user_secrets_rename.rc == 2
+      failed_when: user_secrets_rename.rc not in [0, 2]
+
+    - name: Copy the generally applicable RPC-O config files
+      copy:
+        src: "{{ rpco_cfg_src_path }}/{{ item.name }}"
+        dest: "/etc/openstack_deploy/{{ item.name | regex_replace('.aio$', '') }}"
+        mode: "{{ item.mode | default(omit) }}"
+      when: "{{ (item.condition | default(True)) | bool }}"
+      with_items:
+        - name: "user_osa_variables_defaults.yml"
+          mode: "0440"
+        - name: "user_rpco_variables_defaults.yml"
+          mode: "0440"
+        - name: "user_rpco_secrets.yml"
+        - name: "env.d/nova.yml"
+        - name: "conf.d/ceph.yml.aio"
+          condition: "{{ rpco_deploy_ceph | bool }}"
+
+    - name: Template the RPC-O config files
+      config_template:
+        src: "{{ rpco_cfg_src_path }}/{{ item.name }}"
+        dest: "/etc/openstack_deploy/{{ item.name | regex_replace('.aio$', '') }}"
+        config_overrides: "{{ item.override | default({}) }}"
+        config_type: "yaml"
+      when: "{{ (item.condition | default(True)) | bool }}"
+      with_items:
+        - name: "user_rpco_variables_overrides.yml.aio"
+        - name: "env.d/ceph.yml"
+          condition: "{{ rpco_deploy_ceph | bool }}"
+          override:
+            container_skel:
+              ceph_osd_container:
+                properties:
+                  is_metal: false
+        - name: "env.d/elasticsearch.yml"
+          condition: "{{ rpco_deploy_elk | bool }}"
+        - name: "env.d/kibana.yml"
+          condition: "{{ rpco_deploy_elk | bool }}"
+        - name: "env.d/logstash.yml"
+          condition: "{{ rpco_deploy_elk | bool }}"
+
+    - name: Remove config files
+      file:
+        path: "/etc/openstack_deploy/conf.d/{{ item.name }}"
+        state: absent
+      when: "{{ (item.condition | default(True)) | bool }}"
+      with_items:
+        - name: "aodh.yml"
+        - name: "ceilometer.yml"
+        - name: "gnocchi.yml"
+        - name: "swift.yml"
+          condition: "{{ not rpco_deploy_swift | bool }}"
+
+    - name: Update the RPC-O secrets
+      shell: "{{ rpco_base_dir }}/scripts/update-secrets.sh"
+
+    - name: Check for the existance of a gating settings file
```

Comment:
We should be able to move the generation of the gating settings file to after /etc/openstack_deploy has been created so that it no longer needs to be written into the repo. Also the placeholder creds should probably be removed.